### PR TITLE
Fix CI problems related to new ps_linklist version

### LINKED ISSUE #23273: CI failing: ps_linklist v4 in integration tests
We have integration tests CI failing on `develop` since 11/02/2021.

We've had many different observable issues related to it.

Example
```
PHP Fatal error:  Uncaught Symfony\Component\DependencyInjection\Exception\InvalidArgumentException: Invalid service "prestashop.module.link_block.form_provider":
method "PrestaShop\Module\LinkList\Form\LinkBlockFormDataProvider::__construct()" has no argument named "$configuration".
Check your service definition. in /home/runner/work/PrestaShop/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/ResolveNamedArgumentsPass.php:65

Stack trace:
#0 /home/runner/work/PrestaShop/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php(60): Symfony\Component\DependencyInjection\Compiler\ResolveNamedArgumentsPass->processValue(Object(Symfony\Component\DependencyInjection\Definition), true)
#1 /home/runner/work/PrestaShop/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/ResolveNamedArgumentsPass.php(32): Symfony\Component\DependencyInjection\Compiler\AbstractRecursivePass->processValue(Array, true)
#2 /home/runner/ in /home/runner/work/PrestaShop/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/ResolveNamedArgumentsPass.php on line 65
Fatal error: Uncaught Symfony\Component\DependencyInjection\Exception\InvalidArgumentException: Invalid service "prestashop.module.link_block.form_provider": method "PrestaShop\Module\LinkList\Form\LinkBlockFormDataProvider::__construct()" has no argument named "$configuration". Check your service definition. in /home/runner/work/PrestaShop/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/ResolveNamedArgumentsPass.php on line 65
```

**This issues is currently preventing the merge of all PRs on `develop` and is consequently a high priority for maintainers.**

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | We updated the `ps_linklist` module in composer because it didn't match the one from Addons which was more recent, so there were conflicts between the one from composer and the installed one<br><br>Another problem was that we use the modules test folder in too many places (especially in CreateTestDb) which caused this duplication of modules with different versions<br>The command that create test db now is based on the real modules folder which prevents many issues<br><br>Additionally a work was done in several services that scan the modules and didn't use the const properly so in test environment they kept using the regular folder
| Type?             | bug fix
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/23273
| How to test?      | CI tests green
| Possible impacts? | ~

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23264)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Comment by matks** [comment]
Still in error 😭 

Also, don't hesitate to explain why the changes in your PR fix the issue. I can see that you do an upgrade of ps_linklist module and change a call to translator in one Product Form Type, but I cannot see (without opening the related code in the IDE) why this solves the solution 😉

**Review by Progi1984 — DISMISSED** [review]

File: tests-legacy/resources/modules/statsbestsuppliers/statsbestsuppliers.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            array(
+                'id' => 'sales',
+                'header' => $this->trans('Total paid', array(), 'Modules.Statsbestsuppliers.Admin'),
+                'dataIndex' => 'sales',
+                'align' => 'center'
+            )
+        );
+
+        $this->displayName = $this->trans('Best suppliers', array(), 'Modules.Statsbestsuppliers.Admin');
+        $this->description = $this->trans('Adds a list of the best suppliers to the Stats dashboard.', array(), 'Modules.Statsbestsuppliers.Admin');
```
Comment (Julievrz) [inline tests-legacy/resources/modules/statsbestsuppliers/statsbestsuppliers.php]:
```suggestion
        $this->description = $this->trans('Enrich your stats, add a list of the best suppliers to the dashboard.', array(), 'Modules.Statsbestsuppliers.Admin');
```

Comment (PierreRambaud) [reply tests-legacy/resources/modules/statsbestsuppliers/statsbestsuppliers.php]: Don't waste your time on wording in this PR, we are only playing with tests :sweat_smile:

File: classes/module/Module.php [hunk 2]
```diff
@@ -3439,6 +3439,10 @@ protected function trans($id, array $parameters = [], $domain = null, $locale =
      */
     public function isUsingNewTranslationSystem()
     {
+        if ($this->context === null || $this->context->getTranslator() === null) {
```
Comment (jolelievre) [inline classes/module/Module.php]: @PierreRambaud are you sure about this one? Should we integrate this modification that you needed for tests weird environment or let it explode as it's not supposed to happen?

Comment (PierreRambaud) [reply classes/module/Module.php]: You can remove it and retry in local. It happened because of local settings maybe.

**Review by PierreRambaud — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**